apps/full-site-editing/full-site-editing-plugin/wpcom-block-editor-nav-sidebar/src/components/nav-sidebar/index.tsx

```diff
@@ -63,6 +62,57 @@ function WpcomBlockEditorNavSidebar() {
 		prevIsOpen.current = isOpen;
 	}, [ isOpen, prevIsOpen, setSidebarClosing ] );
 
+	const [ isListScrolling, setIsListScrolling ] = useState( false );
```

Comment:
Hmm the logic around isListScrolling doesn't appear to be working.
I expected it to be true only when scrolling is in progress
It was true when not scrolling as well, it did briefly switch to false around the end of the scroll.

Also I don't understand the need for the complexity of the code to handle this case, it seems to be only used to apply a border above the create button which I think could be always present to simplify the logic? 

![Jul-16-2020 13-56-35](https://user-images.githubusercontent.com/22446385/87625106-ab25d480-c76c-11ea-9848-350ea1baa63a.gif)

Reply:
> I expected it to be true only when scrolling is in progress

I might be able to name this better. It's supposed to be true when there's not enough room in the sidebar to display the whole list. Maybe `willScroll` is better? Or `isScrollbarPresent`?

Yes, it's unfortunately a whole lot of logic just to apply the border above the create button. Thing is I think it makes a big difference having the border there when the button's stuck to the bottom of the sidebar. And I also think having the border doesn't look great when the button isn't stuck to the bottom. Which is why I think the extra logic is worth it.

Ideally there'd be a CSS only solution for this but I couldn't find one. There's probably also a naive implementation that would just measure the sidebar and list in pixels using `.getBoundingClientRect()`. But that's been slow when I've tried that in the poast. But I believe the new `IntersectionObserver` API is designed to be a more performant API.

Reply:
Ahh I see, yea I was confused about the purpose of it, that name change would help! But yea hmm I do think that there would be a design solution which would be intuitive and attractive and wouldn't require the calculation, but it works and TIL about `IntersectionObserver` 😍